### ISSUE TITLE
libvterm build failure with Ninja due to multiple rules for vterm.lib

### ISSUE THREAD

**Issue by Fuzzier** [neovim/neovim#15085]
- `nvim --version`: v0.6.0, master branch, rev 02bf251bb3f1dd55a2c6599790ae6a7772082cdd
- Operating system/version: Windows 10 Pro x64 21H1 19043.1083
- Terminal name/version: cmd.exe
- `$TERM`: cmd.exe
- Build tool: Visual Studio 2019 v16.10.3, x64

### Build steps

1. Clone `https://github.com/neovim/neovim.git` in to `neovim`.
2. Pull the latest master branch.
3. Start `x64 Native Tools Command Prompt for VS 2019`, or start a console and call `vcvarsall.bat x64` manually.
4. Run the following script to build third-party dependencies.

```shell
REM Goto 'neovim\third-party' subdirectory.
PUSHD neovim\third-party
REM Make 'Release' subdirectory.
MD Release
REM Configure CMake.
cmake -H. -BRelease -G Ninja -DCMAKE_BUILD_TYPE=Release
REM Build third-party dependencies, use 1 thread to see what goes wrong.
cmake --build Release -j 1
POPD
```

5. The build failed with the following message.

```shell
CMake Error:
  Running

   'C:/PROGRA~2/MICROS~2/2019/ENTERP~1/Common7/IDE/COMMON~1/MICROS~1/CMake/Ninja/ninja.exe' '-C' '.../neovim/third-party/Release/build/src/libvterm' '-t' 'recompact'

  failed with:

   ninja: error: build.ninja:293: multiple rules generate vterm.lib [-w dupbuild=err]
```

### The problem

In `LibvtermCMakeLists.txt` line 41-50:

```cmake
add_library(vterm ${VTERM_SOURCES} ${TBL_FILES_HEADERS})
install(TARGETS vterm ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})

add_library(vterm-shared SHARED ${VTERM_SOURCES} ${TBL_FILES_HEADERS})
set_target_properties(vterm-shared PROPERTIES
  OUTPUT_NAME vterm
  SOVERSION 0)
install(TARGETS vterm-shared
  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
```
The `LibvtermCMakeLists.txt` tries to build both *static* and *shared* vterm, and the generated `neovim\third-party\Release\build\src\libvterm\build.ninja` contains multiple rules for building `vterm.lib`, which causes the build to fail.

### Recommended solution

The  `LibvtermCMakeLists.txt` should comment out the *shared* vterm.
```cmake
#add_library(vterm-shared SHARED ${VTERM_SOURCES} ${TBL_FILES_HEADERS})
#set_target_properties(vterm-shared PROPERTIES
#  OUTPUT_NAME vterm
#  SOVERSION 0)
#install(TARGETS vterm-shared
#  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
#  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
```

Then the third-party dependencies are built without problems, and `neovim` can be built with the following script.

```shell
PUSHD neovim
MD Release
cmake -H. -BRelease -G Ninja -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX="C:\neovim" -DDEPS_PREFIX="third-party\Release\usr" -DUSE_BUNDLED=ON -DUSE_EXISTING_SRC_DIR=ON -DUSE_BUNDLED_LUA=ON -DCMAKE_C_FLAGS="/DWIN32 /D_WINDOWS /W3 /utf-8"
cmake --build Release
cmake --build Release --target install
POPD
```

**Comment by spindensity** [comment]
I have the same problem and the "recommended solution" post by @Fuzzier works flawlessly.

**Comment by justinmk** [comment]
@Fuzzier can you send a PR?

**Comment by bb010g** [comment]
Resolved by #20566.

**Comment by teto** [comment]
clsoing then, @Fuzzier feel free to reopen if that's not the case
